pkg/identityManager/identityManager.go

```diff
@@ -8,18 +8,35 @@ import (
 )
 
 type certificateIdentityManager struct {
+	identityProvider
+
 	client           kubernetes.Interface
 	localClusterID   clusterid.ClusterID
 	namespaceManager tenantnamespace.Manager
 }
 
 // NewCertificateIdentityManager gets a new certificate identity manager.
 func NewCertificateIdentityManager(client kubernetes.Interface,
```

Comment:
Would personally split this function in two (one for certificates and the other for iam), to avoid relying on the awsConfig parameter being null, which is difficult to understand when observing only the function call. Also the structure name should be probably generalized to `IdentityManager`.